iroh/src/commands/delete.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::Result;
+use clap::Subcommand;
+use iroh::client::quic::Iroh;
+use iroh_bytes::Hash;
+
+#[derive(Subcommand, Debug, Clone)]
+pub enum Commands {
```

Comment:
shouldn't the delete commands be under their respective existing commands? (or for tag a new one)

Reply:
?

We got the following hierarchy:

blob add
blob download
blob list <what to list>

this adds

blob delete <what to delete>

you want to add a new top level thing for tags, so

tag delete

?

Reply:
yeah, that was my thinking

Reply:
this one is tough. IMHO the question of weather tags should be their own thing or not boils down to **how tags & documents relate.**. Do documents have a tag? It could be either implicit (the docs system won't let you delete stuff docs need, aka "implict tag") or explicit (every document gets a tag when you join). If all documents have a tag, then I think tags should be a top-level concept.

But if that's true,  I think we should take a step back, and consider renaming the concept of `tags` to `names` and just turning this into a petname system. We need a way to name stuff, and stuff we name expresses user intent about cache retention. So let's lean into that and have one concept that says "stuff you name is stuff you keep". If I have two chickens, one named Wilber, and the other one doesn't have a name, you can guess which chicken I would "evict from cache" first.

Reply:
so, I am pretty sure you don't want to pin every single document you have ever interacted with in any way. why should documents get an "everything I contain is important" excemption compared to collections?

So yes, I would think that docs should also be taggable, with the default behaviour for an untagged doc being just like an untagged collection.

> I have two chickens, one named Wilber, and the other one doesn't have a name, you can guess which chicken I would "evict from cache" first.

🐔 🍗 🤣 

I am OK with calling tags names, provided that names are still just bytes. We can make it a convention that you usually use utf-8 strings, but I really don't want to enforce that.

Another problem with names is that people will immediately think about something like IPNS...

Reply:
> I am OK with calling tags names, provided that names are still just bytes.

Yep. Super into that convention, which I think we should use everywhere (collection keys if we do them, document keys, and now tag/name names)

> Another problem with names is that people will immediately think about something like IPNS...

I wouldn't say "everyone", but I think you're right that the word `name` generally evokes the notion of a universal name. Humans generally have 1 "name". **I change my vote back to "tags"**

Reply:
So should we move tags to the top level of the cmd hierarchy in anticipation of them being relevant for docs in the future?